README
SYNOPSIS

     use Perinci::Examples qw(delay);
     delay();

DESCRIPTION

    This module and its submodules contain an odd mix of various functions,
    variables, and other code entities, along with their Rinci metadata.
    Mostly used for testing Rinci specification and the various Perinci
    modules.

    Example scripts are put in a separate distribution (see
    Perinci::Examples::Bin) to make dependencies for this distribution
    minimal (e.g. not depending on Perinci::CmdLine) since this example
    module(s) are usually used in the tests of other modules.

SEE ALSO

    Perinci

    Perinci::Examples::Bin



Changes
Revision history for Perinci-Examples

0.45    2015-01-04 (PERLANCAR)

	- Fix/tweak completion example: add description to each answer
	  entry.


0.44    2014-12-03 (PERLANCAR)

	- Add function: demo_cli_opts_shorter.

	- Tweak Rinci metadata, change summary.alt.neg ->
	  summary.alt.bool.not to comply with DefHash 1.0.8; add
	  summary.alt.numnoun.singular in example.


0.43    2014-11-20 (PERLANCAR)

	- Rename function: funcname -> demo_cli_opts (for better name when
	  being made a CLI script).


0.42    2014-11-20 (PERLANCAR)

	- Add module: Perinci::Examples::CLI (for testing
          Perinci-Sub-To-CLIOptSpec).


0.41    2014-11-12 (PERLANCAR)

	- Missed a function.


0.40    2014-11-12 (PERLANCAR)

	- CmdLineSrc: return raw args in extra result (in result metadata), to
          allow testing for special arguments (without breaking existing tests).


0.39    2014-10-30 (PERLANCAR)

	- Rename module Perinci::Examples::File to
          Perinci::Examples::FilePartial.

	- Add module: Perinci::Examples::FileStream.


0.38    2014-10-29 (PERLANCAR)

	- Add module: Perinci::Examples::File.

	- Add function: multi_status().


0.37    2014-10-24 (PERLANCAR)

	- Add function: test_binary (in Perinci::Examples::CmdLineSrc).


0.36    2014-10-24 (PERLANCAR)

	- Add more explanation about the functions/arguments; add function
	  'gen_random_bytes'.


0.35    2014-10-23 (PERLANCAR)

	- Update Rinci version.


0.34    2014-10-11 (PERLANCAR)

	- Use is_flag property (Rinci 1.1.57).


0.33     2014-10-11 (PERLANCAR)

         - Add argument submetadata examples in Perinci::Examples::SubMeta.


0.32     2014-09-11 (PERLANCAR)

         - Add example to demonstrate cmdline_src=stdin_line and is_password
           (Rinci 1.1.53).

         - Adjust to Perinci::Sub::Wrapper 0.62
           (_perinci.sub.wrapper.validate_args is now renamed).

         - Add function: is_palindrome (another demonstration of
           cmdline.result).


0.31     2014-08-23 (SHARYANTO)

         - Add module: Perinci::Examples::{CmdLineResMeta}. This contains
           functions that set cmdline.* result metadata to be tested when
           testing Perinci::CmdLine & Perinci::CmdLine::Lite.


0.30     2014-08-22 (SHARYANTO)

         - Add modules: Perinci::Examples::{CmdLineSrc,Tx}. These are moved from
           test script for Perinci::CmdLine.


0.29     2014-08-21 (SHARYANTO)

         - Add function: test_dry_run.


0.28     2014-07-31 (SHARYANTO)

         - Add functions: test_args_as_{array,arrayref,hashref}.


0.27     2014-07-29 (SHARYANTO)

         - test_completion(): Fix putting 'greedy' metadata property on
           non-array arg.


0.26     2014-07-29 (SHARYANTO)

         - test_completion(): Add pos/greedy properties to help test
           Perinci::Sub::Complete.


0.25     2014-07-23 (SHARYANTO)

         - Add function: gen_sample_data().


0.24     2014-07-03 (SHARYANTO)

         - Add function: test_common_opts().


0.23     2014-06-26 (SHARYANTO)

         - Add more arguments to test_completion() to test various stuffs.


0.22     2014-06-18 (SHARYANTO)

         - Add a2 argument to test_completion() to test element_completion.


0.21     2014-06-18 (SHARYANTO)

         - Add a1 argument to test_completion() to test complete_arg_elem.


0.20     2014-06-11 (SHARYANTO)

         - Add function: return_args.


0.19     2014-03-15 (SHARYANTO)

         - No functional changes. Rebuild to fix POD documentation (function
           argument documentation was broken in Perinci-To-POD < 0.42).

0.18    2013-12-21 (SHARYANTO)

	- Add some response content to call_randlog().

        - Add function arg_default to test default value from default and/or
          schema.


0.17    2013-11-14 (SHARYANTO)

	- Add module: Perinci::Examples::Completion.


0.16    2013-11-12 (SHARYANTO)

	- No functional changes. Rebuild to fix t/00-compile.t generated by
	  older DZP::Test::Compile (2.019 < x < 2.033).


0.15     2013-10-15 (SHARYANTO)

         - Add function: undescribed_args.

         - sum: Add examples with 'src'.


0.14     2012-12-22 (SHARYANTO)

         - Split scripts (bin/) to Perinci-Examples-Bin to make dependencies for
           this distribution minimal (e.g. not depending on Perinci::CmdLine)
           since Perinci::Examples is used in tests for other modules.


0.13     2012-11-29 (SHARYANTO)

         - Add call_gen_array() and call_randlog().


0.12     2012-11-08 (SHARYANTO)

         - No functional changes. Add some # VALIDATE_ARG and rebuild with
           DZP::Rinci::Validate enabled.


0.11     2012-10-31 (SHARYANTO)

         - rename script: peri-{ex,eg}-test-completion

         - sum_array(): Add examples in metadata and 'round' option (to test
           Test::Rinci).


0.10     2012-07-19 (SHARYANTO)

         - Add function: merge_hash


0.09     2012-07-19 (SHARYANTO)

         - Add function: sum

         - Add script: peri-ex-test-completion

         - Tweak some metadata


0.08     2012-02-17 (SHARYANTO)

         - No functional changes. Fix schema for 'test_completion'.


0.07     2012-02-16 (SHARYANTO)

         - Add function: test_completion.


0.06     2012-02-13 (SHARYANTO)

         - First release of Perinci-Examples. Rename distribution from
           Sub-Spec-Examples.


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Various examples of Rinci metadata",
  "AUTHOR" => "perlancar <perlancar\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Perl::osnames" => "0.09"
  },
  "DISTNAME" => "Perinci-Examples",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "Perinci::Examples",
  "PREREQ_PM" => {
    "Log::Any" => 0,
    "Perinci::Object" => "0.18",
    "Perinci::Sub::Util" => "0.38",
    "Rinci" => "1.1.69",
    "experimental" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.45",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Log::Any" => 0,
  "Perinci::Object" => "0.18",
  "Perinci::Sub::Util" => "0.38",
  "Perl::osnames" => "0.09",
  "Rinci" => "1.1.69",
  "Test::More" => 0,
  "experimental" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



